server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -3486,17 +3481,6 @@ else if (!template.isPublicTemplate() && caller.getType() != Account.ACCOUNT_TYP
                 }
             }
 
-            if (!isIso) {
-                // add hypervisor criteria for template case
-                if (hypers != null && !hypers.isEmpty()) {
-                    String[] relatedHypers = new String[hypers.size()];
-                    for (int i = 0; i < hypers.size(); i++) {
-                        relatedHypers[i] = hypers.get(i).toString();
-                    }
-                    sc.addAnd("hypervisorType", SearchCriteria.Op.IN, relatedHypers);
-                }
-            }
-
```

Comment:
can you make it a method and put the call ...

Reply:
@DaanHoogland  done